src/object/object_primitive.c

```diff
@@ -16399,21 +16399,24 @@ mr_get_compression_length (const char *string, int charlen)
  */
 
 int
-mr_get_size_and_write_string_to_buffer (OR_BUF * buf, char *val_p, DB_VALUE * value, int *val_size, int align)
+pr_get_size_and_write_string_to_buffer (OR_BUF * buf, char *val_p, DB_VALUE * value, int *val_size, int align)
 {
   char *compressed_string = NULL, *string = NULL, *str = NULL;
   int rc = NO_ERROR, str_length = 0, length = 0;
   lzo_uint compression_length = 0;
   lzo_bytep wrkmem = NULL;
   bool compressed = false;
-  int error_abort = 0;
+  int error_abort = -1;
 
   /* Checks to be sure that we have the correct input */
   assert (DB_VALUE_DOMAIN_TYPE (value) == DB_TYPE_VARNCHAR || DB_VALUE_DOMAIN_TYPE (value) == DB_TYPE_STRING);
   assert (DB_GET_STRING_SIZE (value) >= PRIM_MINIMUM_STRING_LENGTH_FOR_COMPRESSION);
 
-  error_abort = buf->error_abort;
-  buf->error_abort = 0;
+  if (buf->error_abort != NULL)
```

Comment:
@lucianlazar2211 

valgrind will complain to access `buf->error_abort` is also UMR. 

Please look at `qdata_copy_db_value_to_tuple_value`. 
`buf` is not initialized for `if` case and reach here. 

I think we have to change callers to initialize `OR_BUF` argument, at least `error_abort`. I don't really like the idea.